File: backend/internal/util/pagination.go
package util

import "fmt"

const (
	// DefaultPage 默认页码
	DefaultPage = 1
	// DefaultPageSize 默认每页数量
	DefaultPageSize = 20
	// MaxPageSize 最大每页数量
	MaxPageSize = 100
	// MinPageSize 最小每页数量
	MinPageSize = 1
)

// ValidatePagination 验证并规范化分页参数
// 返回值：规范化后的page、pageSize、错误信息
func ValidatePagination(page, pageSize int) (int, int, error) {
	// 规范化page
	if page < 1 {
		page = DefaultPage
	}

	// 规范化pageSize
	if pageSize < 1 {
		pageSize = DefaultPageSize
	}

	// 验证pageSize上限
	if pageSize > MaxPageSize {
		return 0, 0, fmt.Errorf("单页最多返回%d条记录", MaxPageSize)
	}

	return page, pageSize, nil
}

// ValidatePaginationInt64 验证并规范化分页参数（int64版本）
func ValidatePaginationInt64(page, pageSize int64) (int64, int64, error) {
	// 规范化page
	if page < 1 {
		page = DefaultPage
	}

	// 规范化pageSize
	if pageSize < 1 {
		pageSize = DefaultPageSize
	}

	// 验证pageSize上限
	if pageSize > MaxPageSize {
		return 0, 0, fmt.Errorf("单页最多返回%d条记录", MaxPageSize)
	}

	return page, pageSize, nil
}

// CalculateOffset 计算偏移量（用于数据库查询）
func CalculateOffset(page, pageSize int) int {
	if page < 1 {
		page = 1
	}
	return (page - 1) * pageSize
}

// CalculateOffsetInt64 计算偏移量（int64版本）
func CalculateOffsetInt64(page, pageSize int64) int64 {
	if page < 1 {
		page = 1
	}
	return (page - 1) * pageSize
}

File: frontend/src/constants/pagination.ts
/**
 * 分页常量配置
 * 统一前端分页参数，与后端对齐
 */

export const PAGINATION = {
  DEFAULT_PAGE: 1,
  DEFAULT_PAGE_SIZE: 20,
  MAX_PAGE_SIZE: 100,
  MIN_PAGE_SIZE: 1,
} as const

export const PAGE_SIZE_OPTIONS = [10, 20, 50, 100] as const

/**
 * 标准化分页参数
 * @param page 页码
 * @param pageSize 每页数量
 * @returns 标准化后的分页参数
 */
export function normalizePagination(page?: number, pageSize?: number) {
  const normalizedPage = Math.max(page || PAGINATION.DEFAULT_PAGE, PAGINATION.DEFAULT_PAGE)
  const normalizedPageSize = Math.max(
    Math.min(pageSize || PAGINATION.DEFAULT_PAGE_SIZE, PAGINATION.MAX_PAGE_SIZE),
    PAGINATION.MIN_PAGE_SIZE
  )
  
  return {
    page: normalizedPage,
    page_size: normalizedPageSize,
  }
}
